iceoryx_posh/source/roudi/roudi_process.cpp

```diff
@@ -278,23 +419,33 @@ bool ProcessManager::removeProcess(const ProcessName_t& name) noexcept
     auto it = m_processList.begin();
     while (it != m_processList.end())
```

Comment:
This should be a ranged based loop like
`for(auto & process : m_processList) {...`
Which would reduce the code and would make it even more readable

Reply:
nope, the list is mutated in the loop

Reply:
@elBoberido @mossmaurice by definition a range based loop like:
`for(auto & element : list)` is expanded to
```cpp
  auto && __range = range_expr ; 
  for (auto __it = begin(__range), __e = end(__range); __it != __e; ++__it) { 
    element_decl = *it; 
    statement 
  } 
```

Which is in the end the construct we are using here.
In the end you could translate this while loop one to one into:
```cpp
for(auto it = m_processList.begin(); it != m_processList.end(); ++it)
```

Which is exactly what a range based for loop would expand to according to the standard.
Take a look at the technical explanation: https://en.cppreference.com/w/cpp/language/range-for
And the more user friendly one: https://arne-mertz.de/2015/07/new-c-features-stdbeginend-and-range-based-for-loops/

Reply:
@elBoberido I think you confused it with a vector. If a vector is modified then the iterator become invalid but a list can be modified without invalidating the iterator.

Reply:
So, you're saying we can also use a range based loop?!
But we don't want to do a  `++it` after each iteration but `auto it = m_processList.begin();` instead.

Reply:
@elfenpiff I would even accept a slap bet that your proposal results in a segmentation fault ;)

Reply:
@elBoberido you are right this would lead to a segfault ...

Reply:
So, this one can be dismissed?!

Reply:
@elfenpiff no worries
@marthtz yep